sdk/iot_hub/src/service/mod.rs

```diff
@@ -116,10 +90,14 @@ impl ServiceClient {
             iot_hub_name, &expiry_date_seconds
         );
 
-        let key = decode(private_key).map_err(GenerateSasTokenError::DecodePrivateKeyError)?;
+        let key = decode(private_key).with_context(ErrorKind::Other, || {
```

Comment:
Would it be appropriate for errors relating to the construction of a SAS token to be `ErrorKind::Credential` instead of `ErrorKind::Other`?